MakeHist/mumu.C

```diff
@@ -1,21 +1,111 @@
 #define mumu_cxx
 #include "mumu.h"
-#include <TH2.h>
 #include <TStyle.h>
 #include <TCanvas.h>
-#include <TVector2.h>
 #include <vector>
-#include <TH1.h>
 #include <TLorentzVector.h>
 #include <iostream>
+#include <TVector2.h>
 
 struct Lepton{
     TLorentzVector LV;
     float Iso;
 };
 
+void BinLogX(TH1*h) 
+{
+
+   TAxis *axis = h->GetXaxis(); 
+   int bins = axis->GetNbins();
+
+   Axis_t from = axis->GetXmin();
+   Axis_t to = axis->GetXmax();
+   Axis_t width = (to - from) / bins;
+   Axis_t *new_bins = new Axis_t[bins + 1];
 
-void mumu::Loop(TString sample_name, Float_t xsection, Float_t targetLumi, Float_t numberOfEvents, TString caseText)
+   for (int i = 0; i <= bins; i++) {
+     new_bins[i] = TMath::Power(10, from + i * width);
+   } 
+   axis->Set(bins, new_bins); 
+   delete new_bins; 
+}
+
+float calculateHTLT(vector<TLorentzVector> bJets, vector<TLorentzVector> non_bJets, vector<TLorentzVector> leptons)
+{
+    //(HT-LT) calculation
+    float DHTLT = -999.;
+    //check leptons collection size, throw a warning if it is less than 2
+    if (leptons.size()<2) { std::cout << "HTLT computation: WARNING! Lepton collection size is less than 2!!!" << std::endl; return DHTLT;}
+    //sort leptons collection
+    std::sort(leptons.begin(), leptons.end(), [](TLorentzVector a, TLorentzVector b){return a.Pt() > b.Pt();});
+
+    if(bJets.size() == 1){
+        if (non_bJets.size() == 0){
+            DHTLT = (bJets[0].Pt()) - (leptons[0].Pt()+leptons[1].Pt());
+        } else {
+            std::sort(non_bJets.begin(), non_bJets.end(), [](TLorentzVector a, TLorentzVector b){return a.Pt() > b.Pt();});
```

Comment:
Our selections have no case in which you'd need the sorting. Same comment applies to line 51.
You left it out in the no b-jets case, anyways. That's inconsistent, but no big deal.

Reply:
It rather doesn't matter whether they are pT-ordered there, as you take every single hadronic object in the events in all cases we actually use.